src/api/include/migraphx/migraphx.hpp

```diff
@@ -469,20 +525,29 @@ struct interface_base : Base
     template <class T, class U>
     void auto_assign(rank<0>, T* out, U x)
     {
-        return *out = x;
+        *out = x;
     }
 
     template <class T, class U>
     auto auto_assign(rank<1>, T* out, U x) -> decltype(x.assign_to_handle(out))
     {
         x.assign_to_handle(out);
     }
+
+    template <class T1, class T2, class U, class = std::enable_if_t<std::is_same<T2, size_t>{}>>
+    auto auto_assign(rank<2>, T1* out_ptr, T2* out_size, U x)
+    {
+        std::copy_n(x.begin(), x.size(), out_ptr);
```

Comment:
If `out_ptr` is smaller than `x.size()` then this will write pass the end of the buffer. I would assume `out_size` would be the size of the buffer(passed by value instead of pointer), but then we would need a way to communicate the size back to as well so its probably better as a pointer.

Instead, we probably need to have `out_size` set to the size of `out_ptr` and then later we set it to the smaller size used:

```cpp
template <class T1, class T2, class U, class = std::enable_if_t<std::is_same<T2, size_t>{}>>
auto auto_assign(rank<2>, T1* out_ptr, T2* out_size, U x)
{
    *out_size = std::min(*out_size, x.size());
    std::copy_n(x.begin(), *out_size, out_ptr);
}
```

Reply:
Done.